xbmc/filesystem/AudioBookFileDirectory.cpp

```diff
@@ -75,6 +75,53 @@ bool CAudioBookFileDirectory::GetDirectory(const CURL& url,
   if (m_fctx->nb_chapters > 1)
     thumb = IMAGE_FILES::URLFromFile(url.Get(), "music");
 
+  //Look for any embedded art
+  size_t size = 0;
+  uint8_t* pic{};
+  std::string type;
+
+  for (size_t i = 0; i < m_fctx->nb_streams; ++i)
+  {
+    if ((m_fctx->streams[i]->disposition & AV_DISPOSITION_ATTACHED_PIC) == 0)
+      continue;
+    AVDictionaryEntry* avtag;
+    avtag = av_dict_get(m_fctx->streams[i]->metadata, "filename", nullptr, AV_DICT_IGNORE_SUFFIX);
+    std::string value;
+    if (avtag)
+      value = avtag->value;
+    avtag = av_dict_get(m_fctx->streams[i]->metadata, "mimetype", nullptr, AV_DICT_IGNORE_SUFFIX);
+    if (!value.empty() && avtag)
+    {
+      if (value == "fanart.png" || value == "fanart.jpg")
+        type = "fanart";
+      else if (value == "cover.png" || value == "cover.jpg")
+        type = "thumb";
+      else if (value == "small_cover.png" || value == "small_cover.jpg")
+        type = "thumb";
```

Comment:
type is declared outside the loop and is not reset at the top of the loop. if it's allowed to leak to the next iterations on purpose it should be explained in a comment.

Reply:
I've added a `break` to make sure we don't loop after any art is found.  So either type will be set once, or not at all.